### ISSUE TITLE
Can't display data without tons of labels

### ISSUE THREAD

**Issue by redijedi** [apertureless/vue-chartjs#297]
### Expected Behavior

Add dataset with many points of irregular intervals. Have graph display points between tick labels.

### Actual Behavior
Data from the dataset will only be displayed if there is a label for it. Otherwise it will draw unusual shapes.

I want to show values changing over 365 days. My dataset contains point values. My labels are only for every week, 52 in all. When my points x Val doesn't fall on some labeled tick, things get crazy.

### Environment
- vue.js version: <version here>
- vue-chart.js version: <version here>
- npm version: <version here>

**Comment by apertureless** [comment]
Not sure if I understand you correclty,
However chart.js requires that each data have a corresponding label. Because they get mapped by the index.

So if you have less labels than data, your chart will break.

**Comment by redijedi** [comment]
I've found that leaving out labels from the datasets is a good solution so long as the data isn't reactive. When the data isn't reactive the chart displays as expected. Note the drop from 200 to 160 deg on this chart:

<img width="881" alt="screen shot 2018-01-25 at 2 45 27 pm" src="https://user-images.githubusercontent.com/83999/35408831-85e874da-01de-11e8-9c23-bceb31ac7dd7.png">

Adding the reactive mixin makes it only display one value:

<img width="878" alt="screen shot 2018-01-25 at 2 48 31 pm" src="https://user-images.githubusercontent.com/83999/35408957-dcd8d208-01de-11e8-9039-3d735bdc6283.png">

Also note that the axes labels disappeared. The *only* change here is that the reactive mixin was added as described here: [http://vue-chartjs.org/#/home?id=example](http://vue-chartjs.org/#/home?id=example)

I've found that adding in the `labels` property on the `datasets` makes things look ok so long as you have a label for each possible value. If you don't, things get ugly:

<img width="877" alt="screen shot 2018-01-25 at 2 51 50 pm" src="https://user-images.githubusercontent.com/83999/35409157-5c275a5c-01df-11e8-9cc6-5e4a2435b69c.png">

The workaround is to not use the `reactiveProp` mixin, but I'm guessing that will limit how useful this is when adjusting data from the UI. What I would like to see is the reactive mixin produce a chart that behaves the same way as a non-reactive one. Seems like it currently violates the law of least surprise. It's very surprising that making the graph reactive breaks so many things.

**Comment by apertureless** [comment]
Ah I see.
Well it's not that suprising. Because chartjs does not have a reactive functionality and the mixins are custom made. And have a lot of limitiations, because it is hard to watch that complex objects in all possible variations and usecases.

And the code is rather complex https://github.com/apertureless/vue-chartjs/blob/develop/src/mixins/reactiveProp.js

If only data changes you can call the `update()` method which is implemented in chartjs. But in certain scenarios like adding datasets you have to re-render.

Long story short: 
If you only change your data or add new data, you can add a watcher to your chartData and call `this.$data._chart.update()` and it will be reactive.
